File: Data Structures & Algorithms/two-integer-sum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const numsHash = {};
        const result = [];

        for (const numIndex in nums) {
            const difference = target - nums[numIndex];

            if (numsHash[difference] === undefined) {
                numsHash[nums[numIndex]] = numIndex;
            } else {
                result.push(Number(numsHash[difference]));
                result.push(Number(numIndex));
                break;
            }
        }

        result.sort();
        return result;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, len(nums) - 1

        # while l <= r:
        #     mid = (l + r) // 2

        #     if 

        if nums[l] < nums[r]:
            return nums[l]
        else:
            val = nums[r]

            while r >= 0 and nums[r - 1] < val:
                r -= 1
            
            return nums[r]
